core.lua

```diff
@@ -310,11 +310,22 @@ do
 		if type(raw) ~= "string" then
 			return
 		end
-		local level = raw:match("%+%s*(%d+)")
-		if not level then
+		local regexesFindLevel = { "%+%s*(%d+)", "(%d+)%s*%+", "(%d+)" }
+
+		local level = 0;
+		for i, regex in ipairs(regexesFindLevel) do
+			level = raw:match(regex);
+			level = tonumber(level)
+			if level and level < 32 then
+				break
+			end
+		end
+
+		if not level or level < 1 then
```

Comment:
Since M+ only begins at 2 we could skip if level < 2 here

Reply:
Yeah, I mainly put it to avoid 0 value with the tonumber, but can be use for that as well.